src/compiler/checker.ts

```diff
@@ -33101,10 +33101,11 @@ namespace ts {
                                 Diagnostics.await_expressions_are_only_allowed_at_the_top_level_of_a_file_when_that_file_is_a_module_but_this_file_has_no_imports_or_exports_Consider_adding_an_empty_export_to_make_this_file_a_module);
                             diagnostics.add(diagnostic);
                         }
-                        if ((moduleKind !== ModuleKind.ES2022 && moduleKind !== ModuleKind.ESNext && moduleKind !== ModuleKind.System && !(moduleKind === ModuleKind.NodeNext && getSourceFileOfNode(node).impliedNodeFormat === ModuleKind.ESNext)) || languageVersion < ScriptTarget.ES2017) {
+                        if ((moduleKind !== ModuleKind.ES2022 && moduleKind !== ModuleKind.ESNext && moduleKind !== ModuleKind.System && !(moduleKind >= ModuleKind.Node16 && getSourceFileOfNode(node).impliedNodeFormat === ModuleKind.ESNext)) || languageVersion < ScriptTarget.ES2017) {

```

Comment:
Normally we still emit warnings of some kind when using experimental features, which would still be in all stable versions of node, at least until https://github.com/nodejs/node/pull/42875 ships in the next version of node 18.

Reply:
So maybe we continue erroring and until TLA becomes non-experimental, we could issue an error like

> Top-level 'await' is currently experimental in Node.js. It can only be enabled when the 'module' option is set to 'nodenext'.

Reply:
On a related note, we should probably _remove_ the experimental warning for json imports in nodenext. Afaik, they've stabilized in node 18. At least the version _with_ the import assertion, anyway. It's contentious among the node maintainers.

Reply:
The PR is already merged - that seems like a pretty strong sign that it will be stable, right?

Reply:
The test wasn't wrong, it was just very confusing because the code was in a CJS file. I've updated the check there.

Reply:
At what point do they stop backporting features like that? Maintenance LTS? Is it practical to not create a named moduleResolution target until it solidifies?

Reply:
Changes are backported to LTS versions unless they're somehow breaking from what I understand, while maintenance LTS versions only get security patches. At least I think that's the general policy.